### ISSUE TITLE
Inklecate v0.8.2 compiles invalid JSON

### ISSUE THREAD

**Issue by cproctor** [cproctor/unfold_studio#82]
< ~hp_percent = (player_health / max_health)
    < ~mob_percent = (health / mob_max)
    < {hp_percent}
    ---
    > ~hp_percent = (player_health / (max_health + 0.0001))
    > ~mob_percent = (health / (mob_max + 0.00001))

The number 0.0001 got written by the Ink compiler in scientific notation. It should be 1E-05, but for some reason it was saved instead as  1E-05.0, which is not valid in the format called JSON which it gets saved into. This is an error in the Ink compiler, which is developed by a company called Ink. (Here’s the project https://github.com/inkle/ink) I’m not using their very latest release, so I’m going to see if it’s still a problem in the latest release. If so, I’ll report it as an error. 

I also did some testing on Unfold Studio, and found that only really big or really small numbers get saved into scientific notation, causing this problem. This story runs fine: 

    VAR x = 0
    {x + 0.0001}

but this story crashes:

    VAR x = 0
    {x + 0.00001}

The solution is to either upgrade the inklecate version (check for other errors!) or open a bug in Inklecate if it's not fixed in later versions.

**Comment by cproctor** [comment]
Confirmed that on Ink v0.9 this bug is fixed. Need to bump the internal inklecate version, but first test.

**Comment by cproctor** [comment]
Requires inkjs version bump too, https://github.com/y-lohse/inkjs/issues/355. This is in progress but not yet released.

**Comment by cproctor** [comment]
Once this is in place, re-implement `continue` too, with a maximum step count.

**Comment by cproctor** [comment]
No longer relevant: inklecate has been updated since this was filed; the scientific-notation JSON bug is a compiler defect fixed upstream. Closing.
